runtimes/runtimes/base-runtime.ts

```diff
@@ -116,8 +116,10 @@ export const baseRuntime = (connections: { reader: MessageReader; writer: Messag
             copyFile: (_src, _dest, _options?) => Promise.resolve(),
             exists: _path => Promise.resolve(false),
             getFileSize: _path => Promise.resolve({ size: 0 }),
-            getServerDataDirPath: _serverName => '',
+            getServerDataDirPath: serverName =>
+                lspRouter.clientInitializeParams?.initializationOptions?.aws?.clientDataFolder ?? `/${serverName}`,
```

Comment:
for consistency, I would prefer the logic that we use for `getServerDataDirPath` in standalone: join `clientDataFolder` and `serverName`

Reply:
The idea of this method is that it returns server-specific directory.

Reply:
That makes sense, I changed it to join the paths and use the homedir as the default to be consistent with `getServerDataDirPath` in standalone.